src/vs/workbench/services/themes/common/colorThemeData.ts

```diff
@@ -829,11 +828,12 @@ function nameMatcher(identifers: string[], scope: ProbeScope): number {
 	if (scope.length < identifers.length) {
 		return -1;
 	}
+	const minimumLastScopeIndex = scope.length - identifers.length;
 	let lastScopeIndex = scope.length - 1;
 	let lastIdentifierIndex = findInIdents(scope[lastScopeIndex--], identifers.length);
 	if (lastIdentifierIndex >= 0) {
 		const score = (lastIdentifierIndex + 1) * 0x10000 + identifers[lastIdentifierIndex].length;
-		while (lastScopeIndex >= 0) {
+		while (lastScopeIndex >= minimumLastScopeIndex) {
```

Comment:
@aeschli, I am not 100% certain of this change. On line 828 there is `if (scope.length < identifers.length) {` which seems to indicate that it's ok for the `scope` array to be longer than the `identifiers` array. However, without my change, the condition on 838 is hit and -1 is returned if the `scope` array is longer than `identifiers`.